plot_radial_build.py

```diff
@@ -0,0 +1,119 @@
+import matplotlib.pyplot as plt
+from matplotlib.patches import Rectangle
+import matplotlib.colors
+
+def build_composition_string(composition):
+    """
+    Assembles string from composition dict for use in radial build plot
+
+    Arguments:
+        composition (dict): "material name (str)":volume_fraction (float)
+
+    Returns:
+        comp_string (string): formatted string with material definition
+    """
+
+    comp_string = ''
+    for material, fraction in composition.items():
+        comp_string += f'{material}: {round(fraction*100,3)}%, '
+
+    comp_string = comp_string[0:-2]
+
+    return comp_string
+
+def plot_radial_build(build, Title = "Radial Build", colors = None, height = 20,
+                      max_thickness = None, size = (8,4), unit = 'cm'):
+    """
+    Creates a radial build plot, with layers scaled between a minimum and
+        maximum pixel width to preserve readability
+    
+    Arguments:
+        build (dict): {"layer name": {"thickness": (float),
+                                      "composition": {
+                                        "material name": fraction (float)
+                                }
+                            }
+                        }
+        title (string): title for plot and filename to save to
+        colors (list of str): list of matplotlib color strings. 
+            If specific colors are desired for each layer they can be added here
+        height (float): height to make the rectangles
+        max_thickness (float): maximum thickness of layer to display, useful
+            for reducing the total size of the figure.
+        size (iter of float): figure size, inches. (width, height)
+        unit (str): Unit of thickness values
+    """
+    
+    if colors is None: 
+        colors = list(matplotlib.colors.XKCD_COLORS.values())[0:len(build)]
+
+    #initialize list for lower left corner of each layer rectangle
+    ll = [0,0]
+    plt.figure(1, figsize=size)
+    plt.tight_layout()
+    ax = plt.gca()
+    ax.set_ylim(0,height+1)
+
+    total_thickness = 0
+    for layer, color in zip(build, colors):
+
+        comp_string = build_composition_string(build[layer]['composition'])
+        newlines = comp_string.count('\n')
```

Comment:
Isn't this always 0? I don't see where `build_composition_string` ever adds '\n'?

Reply:
I see now that they may be in the user-supplied data.  Can we come up with a reliable way to do this without the user's guidance?

Reply:
added a method that relies on a scaling factor to go from number of characters to units of rectangle length, whatever they are. I haven't been able to find a good method to get matplotlib to do this for me. The default wrap feature wraps next to fit in the whole plot, not just the rectangle